types/react-instantsearch-core/index.d.ts

```diff
@@ -748,37 +748,83 @@ export type InsightsClient = (method: InsightsClientMethod, payload: InsightsCli
 export type InsightsClientMethod = 'clickedObjectIDsAfterSearch' | 'convertedObjectIDsAfterSearch';
 
 export interface InsightsClientPayload {
-    index: string;
-    queryID: string;
-    eventName: string;
-    objectIDs: string[];
-    positions?: number[] | undefined;
+  index: string;
+  queryID: string;
+  eventName: string;
+  objectIDs: string[];
+  positions?: number[] | undefined;
 }
 
 export type WrappedInsightsClient = (method: InsightsClientMethod, payload: Partial<InsightsClientPayload>) => void;
 export interface ConnectHitInsightsProvided {
-    hit: Hit;
-    insights: WrappedInsightsClient;
+  hit: Hit;
+  insights: WrappedInsightsClient;
 }
 
 export function EXPERIMENTAL_connectConfigureRelatedItems(
   Composed: React.ComponentType<any>
 ): React.ComponentClass<any>;
 export function connectQueryRules(Composed: React.ComponentType<any>): React.ComponentClass<any>;
 export function connectHitInsights(
-    insightsClient: InsightsClient,
+  insightsClient: InsightsClient,
 ): (
-    hitComponent: React.ComponentType<any>,
+  hitComponent: React.ComponentType<any>,
 ) => React.ComponentType<Omit<ConnectHitInsightsProvided, { insights: WrappedInsightsClient }>>;
 export function connectVoiceSearch(Composed: React.ComponentType<any>): React.ComponentClass<any>;
 
-export interface DynamicWidgetsProps  {
-    children: React.ReactNode;
-    attributesToRender: string[];
-    fallbackComponent?: React.ComponentType<{ attribute: string }>;
+export type DynamicWidgetComponent = React.Component<{
+  children?: DynamicWidgetComponent;
+  attribute?: string;
+  attributes?: string[];
+}>;
```

Comment:
I'm not sure about this new type. I added it because we check for `props.attribute`, `props.attributes` and `props.children` in `getAttribute` function in the tests file.